### ISSUE TITLE
Crash on macOS with Apple Silicon during Deduplication Process

### ISSUE THREAD

**Issue by ylluminate** [sahib/rmlint#646]
**Environment:**
- Installed via Homebrew: rmlint Version: 2.10.2 compiled: Sep 28 2023, "Ludicrous Lemur" (rev unknown)
- OS: macOS 14.3
- Shell: zsh 5.9
- Hardware: MacBookPro18,2 (Apple Silicon: M1 Max)

**Description:**
I encountered a crash while using `rmlint` to deduplicate files between two large folders. The process seemed to progress normally but terminated unexpectedly with a malloc error.

**Command Executed:**
```bash
sudo rmlint --keep-all-tagged --must-match-tagged --hidden --no-crossdev --merge-directories --xattr --match-extension --progress --threads=32 /path/to/delete // /path/to/keep
```

**Output and Error:**
The operation proceeded without issues initially, traversing and preprocessing files, and matching duplicates. However, either during the script generation or during the "Merging files into directories" stage, `rmlint` crashed with the following output:

```
▕░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░▏                                                                                      Traversing (17835721 usable files / 0 + 0 ignored files / folders)
▕░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░▏                                                                                      Preprocessing (reduces files to 13012543 / found 28410 other lint)
▕░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░▏                                                                         Matching (4118222 dupes of 3183325 originals; 0 B to scan in 0 files, ETA:  0s)
▕░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░▏                                                                                                            Merging files into directories (stand by...)

==> In total 17835721 files, whereof 4118222 are duplicates in 3183325 groups.
==> This equals 458.17 GB of duplicates which could be removed.
==> 28410 other suspicious item(s) found, which may vary in size.
==> Scanning took in total 58m 37.912s.

Wrote a json file to: /path/to/keep/rmlint.json
Wrote a sh file to: /path/to/keep/rmlint.sh
rmlint(53389,0x1e266dc40) malloc: *** error for object 0x600f4cd65fb0: pointer being freed was not allocated
rmlint(53389,0x1e266dc40) malloc: *** set a breakpoint in malloc_error_break to debug
Aborting due to a fatal error. (signal received: Abort trap: 6)
Please file a bug report (See rmlint -h)
[1]    53388 abort      sudo rmlint --keep-all-tagged --must-match-tagged --hidden --no-crossdev
```

I received a message suggesting filing a bug report. Please let me know if any additional information is required to diagnose this issue.
